code/datums/components/barber.dm

```diff
@@ -0,0 +1,605 @@
+#define HAIRCUT 1
+#define SHAVE 2
+#define BARBERY_FAILURE 4	// if barbering is not successful and does not display a message
+#define BARBERY_SUCCESSFUL 1 // if barbering is successful, don't attack em
+#define BARBERY_RESOLVABLE 2 // if barbering is not successful, but gives a message
+
+/datum/component/toggle_tool_use
+/datum/component/toggle_tool_use/Initialize()
+	if(!istype(parent, /obj/item))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, list(COMSIG_ITEM_DROPPED), .proc/on_drop)
+	RegisterSignal(parent, list(COMSIG_ITEM_ATTACK_SELF), .proc/on_attackself)
+
+	// this proc is supposed to make certain tools less accidentally deadly for inexperienced players to use
+	// when force_use_as_tool is set, all intents will try to do their tool-thing, and if it can't, return a message saying they're using it wrong
+	// if not set, help intent will still attempt tool, but you'll shank them if it doesn't work out
+/datum/component/toggle_tool_use/proc/on_drop(var/obj/item/thing)
+	toggle_force_use_as_tool(thing)
```

Comment:
Changing behaviour based on whether you supply the `user` argument is really unintuitive. Especially considering that this could just be replaced with
```suggestion
	thing.force_use_as_tool = 0
	thing.dir = SOUTH